packit_service/worker/events/new_hotness.py

```diff
@@ -148,3 +147,58 @@ def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
         result = super().get_dict(d)
         result.pop("_repo_url")
         return result
+
+
+# the decorator is needed in case the DB project event is not created (not valid arguments)
+# but we still want to report from pre_check of the PullFromUpstreamHandler
+@use_for_job_config_trigger(trigger_type=JobConfigTriggerType.release)
+class NewHotnessUpdateEvent(AnityaUpdateEvent):
+    def __init__(
+        self,
+        package_name: str,
+        version: str,
+        distgit_project_url: str,
+        bug_id: int,
+        release_monitoring_project_id: int,
+    ):
+        super().__init__(
+            package_name=package_name,
+            distgit_project_url=distgit_project_url,
+            release_monitoring_project_id=release_monitoring_project_id,
+        )
+        self._version = version
+        self.bug_id = bug_id
+
+    @property
+    def version(self) -> str:
+        return self._version
+
+
+@use_for_job_config_trigger(trigger_type=JobConfigTriggerType.release)
+class AnityaVersionUpdateEvent(AnityaUpdateEvent):
+    def __init__(
+        self,
+        package_name: str,
+        versions: list[str],
+        distgit_project_url: str,
+        release_monitoring_project_id: int,
+    ):
+        super().__init__(
+            package_name=package_name,
+            distgit_project_url=distgit_project_url,
+            release_monitoring_project_id=release_monitoring_project_id,
+        )
+
+        self._versions = versions
+
+    @property
+    def version(self) -> str:
+        # TODO: Handle here or further down the chain? we should be able to get
+        # the package config from dist-git and resolve the next version; how are
+        # we going to choose the version?
+        #   a) latest greatest
+        #   b) next unreleased?
+        # this can be also influenced by the mask…
```

Comment:
Example of a problem:

* There's a `1.0.0` release in Fedora.
* We've got releases `1.0.1`, `1.1.0` and `2.0.0` from Anitya
* Which release do we sync? The next possible (`1.0.1`) or latest-greatest `2.0.0`? Syncing all of them would create a lot of noise.
* Ideal scenario here would be having the *version mask* defined, so it would be as explicit as possible.